bees should not attack apids

KazooBard opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request
Bees do not attack apids. I got permission from maintainer PowerfulBacon.
<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game
This fixes the bug
<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->
Also I am not 100% sure if my code will not pull a coconut jpg from tf2 on the servers and break everything due to two line of code being added but I have high hopes for this one. I deleted one empty line separating things and added two lines of code. I tested the basic stuff and bees don't attack apids and do attack humans etc. 
## Changelog
:cl:
fix: apids are no longer attacked by bees
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

KazooBard committed:
bees should not attack the bees

PowerfulBacon commented:
👍

Archanial commented on code/modules/hydroponics/beekeeping/beebox.dm:
```diff
@@ -18,6 +18,8 @@
 /mob/living/carbon/human/bee_friendly()
 	if(dna?.species?.id == "pod") //bees pollinate plants, duh.
 		return 1
+	if(dna?.species?.id == "apid") //bees don't attack bees, why would they?
```
pretty sure there is a helper for that
`isapid()`

KazooBard replied:
I am having problems with the isapid() helper, do you think I can stick with the species id check?

PowerfulBacon replied:
Way easier would be to give apids TRAIT_BEEFRIEND as an inherent trait

yorii replied:
I suggest changing line 19 to:
if(isapid(src) || ispodperson(src))

yorii replied:
try something like this
![image](https://user-images.githubusercontent.com/7960287/116946824-73483900-ac7b-11eb-8a9c-73b4acf2abb2.png)

yorii replied:
Another solution would be to just use the already existing TRAIT_BEEFRIEND and just give that to both apids and podpeople.

Archanial commented on code/modules/hydroponics/beekeeping/beebox.dm:
```diff
@@ -18,6 +18,8 @@
 	if(dna?.species?.id == "pod") //bees pollinate plants, duh.
 		return 1
+	if(dna?.species?.id == "apid") //bees don't attack bees, why would they?
+		return 1
```
```suggestion
		return TRUE
```

KazooBard committed:
Changed the targeting in the bees themselves

I changed it in the code of the bees instead of the apiaries, since I think this is much easier to find later and clean up.

The thought was for it to check if the bees try to attack apid, and if they do they do bee friendly instead of do bee friendly but booled.

PowerfulBacon commented:
Cant you just make apids return true on bee friendly though? (Calling bee_friendly on an apid returns true)

Sinestia commented:
>>This fixes the bug

KazooBard commented:
> 
> 
> Cant you just make apids return true on bee friendly though? (Calling bee_friendly on an apid returns true)

I did that. I made it return true on apids and false on other species (pods are unattacked)>

PowerfulBacon reviewed (changes requested):
The bee_friendly proc should return true for felinids. You shouldn't invert the bee_friendly proc for apids, because if they are bee friendly then they become non bee friendly.

KazooBard commented:
> The bee_friendly proc should return true for felinids. You shouldn't invert the bee_friendly proc for apids, because if they are bee friendly then they become non bee friendly.

What do you mean true for felinids?

PowerfulBacon commented:
I mean apids.

KazooBard committed:
as requested by bacon

KazooBard commented:
> 
> 
> I mean apids.

did it with TRUE instead

PowerfulBacon commented:
you passed true as a parameter, you need to edit the proc to return true if src is an apid

KazooBard committed:
the proc is now returning true for apids

KazooBard commented:
@yorii I will wait for the review of the new changes as they are probably one of the simplest ways to do this. Now I could do it as you asked but I do not know what would it change besides making the line a little shorter. Thanks for the suggestion!

utilbird reviewed (changes requested):
just give TRAIT_BEEFRIEND to apids and podperson in `on_species_gain`, then remove the two species checks

(make sure you remove them in the species remove proc too)

yorii commented:
> just give TRAIT_BEEFRIEND to apids and podperson in `on_species_gain`, then remove the two species checks
> 
> (make sure you remove them in the species remove proc too)

It would seem apids have already had TRAIT_BEEFRIEND for a year now, not sure if this PR is irrelevant or if maybe the original code doesn't work.
